packages/services/api/src/modules/organization/providers/organization-access-tokens.ts

```diff
@@ -315,31 +567,533 @@ export class OrganizationAccessTokens {
+              isGranted: boolean;
+            }>;
+          }
+        >;
+        resourceIds: Array<string>;
+      };
+
+      const resourceLevelGroups = new Map<ResourceLevel, PMap>(
+        (
+          [
+            'organization',
+            'project',
+            'target',
+            'service',
+            'appDeployment',
+          ] satisfies Array<ResourceLevel>
+        ).map((value): [ResourceLevel, PMap] => [
+          value,
+          {
+            level: value,
+            permissionGroups: new Map(),
+            resourceIds: resourceIds[value] ?? [],
+          },
+        ]),
+      );
+
+      if (accessToken.projectId) {
+        resourceLevelGroups.delete('organization');
+      }
+
+      for (const pgroup of OrganizationAccessTokensPermissions.permissionGroups) {
+        for (const permission of pgroup.permissions) {
+          const resourceLevel = getPermissionGroup(permission.id);
+          const resourceGroup = resourceLevelGroups.get(resourceLevel);
+
+          if (resourceGroup === undefined) {
+            continue;
+          }
+
+          let group = resourceGroup.permissionGroups.get(pgroup.title);
+
+          if (group === undefined) {
+            group = {
+              title: pgroup.title,
+              permissions: [],
+            };
+            resourceGroup.permissionGroups.set(pgroup.title, group);
+          }
+
+          const isGranted = grantedPermissions.has(permission.id);
+
+          if (includeAll || isGranted) {
+            group.permissions.push({
+              isGranted: grantedPermissions.has(permission.id),
+              permission,
+            });
+          }
+        }
+      }
+
+      return Array.from(resourceLevelGroups.values())
+        .map(
+          resourceGroup =>
+            ({
+              level: resourceLevelToResourceLevelType(resourceGroup.level),
+              title: resourceLevelToHumanReadableName(resourceGroup.level),
+              resolvedPermissionGroups: Array.from(resourceGroup.permissionGroups.values())
+                .map(group => ({
+                  title: group.title,
+                  permissions: group.permissions.map(permission => ({
+                    isGranted: permission.isGranted,
+                    permission: permission.permission,
+                  })),
+                }))
+                .filter(group => (includeAll ? true : group.permissions.length !== 0)),
+              resolvedResourceIds: resourceGroup.resourceIds.length
+                ? resourceGroup.resourceIds
+                : null,
+            }) satisfies GraphQLResolvedResourcePermissionGroupOutput,
+        )
+        .filter(group =>
+          includeAll
+            ? true
+            : group.resolvedPermissionGroups.length !== 0 && group.resolvedResourceIds?.length,
+        );
+    };
+
+  static computeAuthorizationStatements(
+    accessToken: OrganizationAccessToken,
+    membership: OrganizationMembership,
+  ) {
```

Comment:
Function used by access token cache for personal access tokens. I made it static to avoid dependency injection where we do not use graphql-modules...